kernel: bump 6.1 to 6.1.40

graysky2 opened the pull request:
kernel: bump 6.1 to 6.1.40

Removed upstreamed:
	generic/backport-6.1/701-v6.5-net-bgmac-postpone-turning-IRQs-off-to-avoid-SoC-han.patch[1]
	generic/pending-6.1/160-workqueue-fix-enum-type-for-gcc-13.patch[2]
	qualcommax/patches-6.1/0022-v6.5-soc-qcom-mdt_loader-Fix-unconditional-call-to-scm_pa.patch[3]

Manually rebased:
	bcm27xx/patches-6.1/950-0359-xhci-quirks-add-link-TRB-quirk-for-VL805.patch
	bcm27xx/patches-6.1/950-0362-usb-xhci-add-VLI_TRB_CACHE_BUG-quirk.patch
	bcm27xx/patches-6.1/950-0390-usb-xhci-add-a-quirk-for-Superspeed-bulk-OUT-transfe.patch
	bcm27xx/patches-6.1/950-0469-usb-xhci-add-XHCI_VLI_HUB_TT_QUIRK.patch
	bcm53xx/patches-6.1/180-usb-xhci-add-support-for-performing-fake-doorbell.patch

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.40&id=685b57a1221c38ec8b456f968264d2496715820c
2. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.40&id=2d57a1590f4d8c516f5aaf8fd5bb4f52d67275d8
3. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.40&id=33f8dff6e1cbba5c2ec85fa5649c0a759a7e685c

Build system: x86/64
Build-tested: x86/64, filogic/xiaomi_redmi-router-ax6000-ubootmod
Run-tested: x86/64, filogic/xiaomi_redmi-router-ax6000-ubootmod

graysky2 committed:
kernel: bump 6.1 to 6.1.39

Removed upstreamed:
	generic/backport-6.1/803-v6.5-01-leds-trigger-netdev-Recheck-NETDEV_LED_MODE_LINKUP-o.patch[1]
	generic/pending-6.1/143-jffs2-reduce-stack-usage-in-jffs2_build_xattr_subsys.patch[2]
	bcm27xx/patches-6.1/950-0034-drm-bridge-Introduce-pre_enable_upstream_first-to-al.patch[3]
	bcm27xx/patches-6.1/950-0439-nvmem-Use-NVMEM_DEVID_AUTO.patch[4]
	bcm4908/patches-6.1/040-mtd-parsers-refer-to-ARCH_BCMBCA-instead-of-ARCH_BCM.patch[5]
	bcm53xx/patches-6.1/031-v6.5-0003-ARM-dts-BCM5301X-Drop-clock-names-from-the-SPI-node.patch[6]
	bcm53xx/patches-6.1/031-v6.5-0015-ARM-dts-BCM5301X-fix-duplex-full-full-duplex.patch[7]
	mediatek/patches-6.1/351-cpufreq-mediatek-correct-voltages-for-MT7622-and-MT7.patch[8]
	qualcommax/patches-6.1/0008-v6.2-clk-qcom-reset-support-resetting-multiple-bits.patch[9]

Manually rebased:
	bcm27xx/patches-6.1/950-0035-drm-panel-Add-prepare_upstream_first-flag-to-drm_pan.patch
	bcm27xx/patches-6.1/950-0404-drm-panel-panel-ilitek9881c-Add-prepare_upstream_fir.patch
	bcm27xx/patches-6.1/950-0699-Bluetooth-hci_sync-Add-fallback-bd-address-prop.patch
	mediatek/patches-6.1/350-21-cpufreq-mediatek-Add-support-for-MT7988.patch

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.39&id=ab0bd172d6289310a05a0cd15e1432e828d386ae
2. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.39&id=6df680709d901346831ef8f221cc90a42062c526
3. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.39&id=5044e5f2511c9afdf9880d2bb6b9d37dfc345dac
4. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.39&id=e27948f329f7e02591ed1feb9a7710c2ccf89a83
5. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.39&id=ad2928e7f3f6120a0bd18aa1056b3b24068027c5
6. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.39&id=2b55a985727833f37c39911f34096b3fdf2a367d
7. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.39&id=90d4c487cd658b51212eb65ae804ab11af193672
8. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.39&id=7e3ee25e8c7c7be1eacdfc6d9f5f0e550a2af241
9. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.39&id=b20854ef6c4955be3310975a72f02d92cb01d6d4

Build system: x86/64
Build-tested: x86/64
Run-tested: x86/64

Co-authored-by: Hauke Mehrtens <hauke@hauke-m.de>
Co-authored-by: Christian Marangi <ansuelsmth@gmail.com>
Signed-off-by: John Audia <therealgraysky@proton.me>
[ rebase bcm27xx and mediatek patches, refresh commit description ]
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
[ minor fixup to bcm27xx patches, refresh commit description ]
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

graysky2 commented:
@hauke @PolynomialDivision Please feel free to use this as a start, but this one is over my head (bcm27xx and mediatek).

PolynomialDivision commented:
@graysky2 You write in your commit message, that you removed `950-0034-drm-bridge-Introduce-pre_enable_upstream_first-to-al.patch` (that's correct it got upstreamed) but did you forget to commit that removal?

graysky2 commented:
Sorry about that, yes.  Fixed.

hauke commented:
I pushed a commit which refreshes the mediatek and bcm27xx target. Feel free to squash it into the original commit.

graysky2 commented:
@hauke 
> I pushed a commit which refreshes the mediatek and bcm27xx target. Feel free to squash it into the original commit.

I looked at your fork but did not see a new commit.  Can you advise?

graysky2 commented:
ping @hauke - can you share the location of the commit you pushed?

DragonBluep commented:
@graysky2 hauke pushed the patch to your graysky2:6.1 branch in this PR, but you accidently override it by force-push. You can download it from here.
https://github.com/openwrt/openwrt/commit/c3a4498b4b5da7c77c707037b51658bff3113603.patch

graysky2 commented:
Thanks @DragonBluep - I did not see his message until after I did the force push.  I need to get into the habit of something like:
```
git pull --rebase && git push origin +$(git rev-parse --abbrev-ref HEAD)
```

graysky2 commented:
@DragonBluep - how did you manage to locate the commit hash of his push after I did the force push?

DragonBluep commented:
You can click here to enter into the commit page and add `.patch` suffix to the URL to download the patch.
![image](https://github.com/openwrt/openwrt/assets/70847398/18b9cd2d-949c-4e1c-990f-9429eb385d9c)

graysky2 commented:
@hauke - we're getting a build fail on bcm27xx.  I have the 6.1.40 push ready but am thinking we want 6.1.39 to be ironed out first.

Ansuel commented:
@graysky2 took care of it just now ;)

graysky2 commented:
> @graysky2 took care of it just now ;)

Thanks, Christian.  Did you commit only change a description?  I already processed 6.1.40 based on this from a few hours ago.

Ansuel commented:
> > @graysky2 took care of it just now ;)
> 
> Thanks, Christian. Did you commit only change a description? I already processed 6.1.40 based on this from a few hours ago.

i refactored the description a bit and had to fix 2 patch for bcm27xx but in theory you should be able to rebase 6.1.40 on top of this with now problem. Also I fixed the author that was wrong... (no idea what happen there)

Ansuel commented:
> bcm53xx/patches-6.1/180-usb-xhci-add-support-for-performing-fake-doorbell.patch
>         bcm27xx/patches-6.1/950-0359-xhci-quirks-add-link-TRB-quirk-for-VL805.patch
>         bcm27xx/patches-6.1/950-0362-usb-xhci-add-VLI_TRB_CACHE_BUG-quirk.patch
>         bcm27xx/patches-6.1/950-0390-usb-xhci-add-a-quirk-for-Superspeed-bulk-OUT-transfe.patch
>         bcm27xx/patches-6.1/950-0469-usb-xhci-add-XHCI_VLI_HUB_TT_QUIRK.patch

the changed patch doesn't affect usb so it should not be needed

graysky2 commented:
OK it probably makes sense for me to just repeat the run and see if any of the 5 patches I called out above in my edit are different as a function of your work.

graysky2 commented:
> the changed patch doesn't affect usb so it should not be needed

OK :+1:

Ansuel commented:
@graysky2 (just in case you have pending change to push and you have problem with refreshing/rebase, you can create a new branch with your local changes committed. (git checkout -b temp) Then fetch origin (git fetch origin) and reset the 6.1 branch to this new version (git reset --hard origin/6.1) and then cherry-pick or merge the temp branch on top of this (git merge temp))

graysky2 committed:
kernel: bump 6.1 to 6.1.40

Removed upstreamed:
	generic/backport-6.1/701-v6.5-net-bgmac-postpone-turning-IRQs-off-to-avoid-SoC-han.patch[1]
	generic/pending-6.1/160-workqueue-fix-enum-type-for-gcc-13.patch[2]
	qualcommax/patches-6.1/0022-v6.5-soc-qcom-mdt_loader-Fix-unconditional-call-to-scm_pa.patch[3]

Manually rebased:
	bcm27xx/patches-6.1/950-0359-xhci-quirks-add-link-TRB-quirk-for-VL805.patch
	bcm27xx/patches-6.1/950-0362-usb-xhci-add-VLI_TRB_CACHE_BUG-quirk.patch
	bcm27xx/patches-6.1/950-0390-usb-xhci-add-a-quirk-for-Superspeed-bulk-OUT-transfe.patch
	bcm27xx/patches-6.1/950-0469-usb-xhci-add-XHCI_VLI_HUB_TT_QUIRK.patch
	bcm53xx/patches-6.1/180-usb-xhci-add-support-for-performing-fake-doorbell.patch

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.40&id=685b57a1221c38ec8b456f968264d2496715820c
2. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.40&id=2d57a1590f4d8c516f5aaf8fd5bb4f52d67275d8
3. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.40&id=33f8dff6e1cbba5c2ec85fa5649c0a759a7e685c

Build system: x86/64, filogic/xiaomi_redmi-router-ax6000-ubootmod
Build-tested: x86/64, filogic/xiaomi_redmi-router-ax6000-ubootmod
Run-tested: x86/64

Signed-off-by: John Audia <therealgraysky@proton.me>

hauke commented on target/linux/bcm27xx/patches-6.1/950-0390-usb-xhci-add-a-quirk-for-Superspeed-bulk-OUT-transfe.patch:
```diff
@@ -86,11 +86,15 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
+-#define XHCI_ZHAOXIN_HOST	BIT_ULL(48)
++#define XHCI_VLI_SS_BULK_OUT_BUG       BIT_ULL(47)
++#define XHCI_ZHAOXIN_TRB_FETCH	BIT_ULL(48)
++#define XHCI_ZHAOXIN_HOST	BIT_ULL(49)
```
Please move `XHCI_VLI_SS_BULK_OUT_BUG` again to the end

Ansuel commented:
@graysky2 yep much better less patch delta for me you can squash to the .40 commit

hauke commented on target/linux/bcm53xx/patches-6.1/180-usb-xhci-add-support-for-performing-fake-doorbell.patch:
```diff
@@ -107,11 +107,15 @@ it on BCM4708 family.
+-#define XHCI_ZHAOXIN_HOST	BIT_ULL(46)
++#define XHCI_FAKE_DOORBELL     BIT_ULL(45)
++#define XHCI_ZHAOXIN_TRB_FETCH	BIT_ULL(46)
++#define XHCI_ZHAOXIN_HOST	BIT_ULL(47)
```
Please move `XHCI_FAKE_DOORBELL` to the end too.
